DOCS-10898-EDIFACT-2.3-ESB (#1451)
mulesoft/docs-connectors: edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-reference.adoc

| Test indicator | String |  Test indicator digit to be used on Send interchanges. By default, interchanges are not sent as tests. |  |
| Use Supplied Values | Boolean | Whether to generate control numbers when writing or use supplied values for control-segment information such as control numbers. The default behavior is to always generate control numbers when writing. This allows you to instead use the values you have chosen. | false |
| Interchange control number key | String |The key-value pair of the interchange control key and interchange control number.| NONE |
| Message control number key | String | The key-value pair of the message control key and message control number.| NONE |
| Truncate values exceeding the length limits | Boolean | Works when enforce length values is `true`. If `true`, values exceeding maximum length are trimmed. If `false`, the values are written as-is. | false |
|===

== Operations

* Read
* Write
* Write batch

== Read

`<edifact:read>`

> Transform an input stream of EDI text into maps and lists structure of EDI data.

Reviewer: ```suggestion
Merge individual EDIFACT messages created and accumulated over a period of time into a single interchange (UNB).
```

Reviewer: ?

=== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | |x
| Read Content | Binary |  |  `#[payload]` |

> | Target Variable | String |  The name of a variable on which the operation's output is placed. |  |

Reviewer: ```suggestion
| Target Variable | String |  Name of the variable that stores the operation's output. |  |
```

> | Target Value | String |  An expression to evaluate against the operation's output and the outcome of that expression is stored in the target variable. |  `#[payload]` |

Reviewer: ```suggestion
| Target Value | String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  `#[payload]` |
```

|===

=== Output

[cols="30a,70a"]
|===
| Type | Object
|===

=== For Configurations

* config

=== Throws

* EDIFACT:SCHEMA
* EDIFACT:WRITE
* EDIFACT:UNKNOWN
* EDIFACT:PARSE

== Write

`<edifact:write>`

> Transform maps and lists structure of EDI data to EDI text stream.

Reviewer: ```suggestion
Transforms the maps and lists structure of EDI data to EDI text stream.
```

Reviewer: ?

=== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | |x
| Write Content | Object |  |  `#[payload]` |
| Streaming Strategy | repeatable-in-memory-stream
repeatable-file-store-stream
non-repeatable-stream |  Configure to use repeatable streams and their behavior. |  |

> | Target Variable | String |  The name of a variable on which the operation's output is placed. |  |

Reviewer: Same comment as previous. These updates can be made in a future iteration though. Not important right now.

| Target Value | String |  An expression to evaluate against the operation's output and the outcome of that expression is stored in the target variable. |  `#[payload]` |
|===

=== Output

[cols="30a,70a"]
|===
| Type | Binary
|===

=== For Configurations

* config

=== Throws

* EDIFACT:SCHEMA
* EDIFACT:WRITE
* EDIFACT:UNKNOWN
* EDIFACT:PARSE

== Write batch

`<edifact:write-batch>`

> Transform multiple instances of maps and lists structures of EDI data stringed together into a single grouped EDI text stream.

Reviewer: the maps and lists structure?
I'm not sure about this so ignore if it doesn't make sense.

Reviewer 2: Rephrase this as:
Merge individual EDIFACT messages created and accumulated over a period of time into a single interchange (UNB).
